### PR TITLE
feat(java-base): Add openjdk-devel package

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

As decided in Arch meeting on 2024-01-25
For https://github.com/stackabletech/spark-k8s-operator/discussions/340

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [x] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: java-base/Dockerfile [hunk 1]
```diff
@@ -19,6 +19,9 @@ RUN microdnf update && \
     microdnf install \
     # Obviously needed to run Java programs
     java-${PRODUCT}-openjdk-headless \
+    # Includes tools such as jps or jmap (which is e.g. needed by the Spark UI, see https://github.com/stackabletech/spark-k8s-operator/discussions/340)
+    # We consider them usefull enough to justify putting them in all Java images
```
Comment (lfrancke) [inline java-base/Dockerfile:23]:
```suggestion
    # We consider them useful enough to justify putting them in all Java images
```

Comment (sbernauer) [reply java-base/Dockerfile]: Where is the spellcheck action when we need it the most? :D

**Comment by lfrancke** [comment]
This should probably also change the other Dockerfiles which might manually install devel right now including the parameters we added for that

Examples: 
https://github.com/stackabletech/docker-images/blob/main/hadoop/Dockerfile#L17
https://github.com/stackabletech/docker-images/blob/main/conf.py#L63-L64

**Comment by sbernauer** [comment]
Good catch, wasn't ware of them!

File: java-base/Dockerfile (hunk 1 above)
Comment (lfrancke) [reply java-base/Dockerfile]: :)

**Review by lfrancke — APPROVED** [review]

**Pull request merged**